### PR TITLE
733: provide commands for the adobe stock asset

### PR DESCRIPTION
**Pull request by coderimus**
### Description (*)
Provide commands for the Adobe Stock asset module. This approach allowed to make related repositories more light-weighted and easy to implement. Makes the module code structure clear and easy to maintain.

After PR approved will cover with tests AssetRepository + others in the Adobe Stock asset module.

### Fixed Issues (if relevant)
#733 

### Manual testing scenarios (*)
N/A

### CODE REVIEW

File: AdobeStockAsset/Model/AppendAttributes.php [hunk 1]
```diff
@@ -7,12 +7,11 @@
 
 namespace Magento\AdobeStockAsset\Model;
 
-use Magento\MediaGalleryApi\Model\Asset\Command\GetByIdInterface;
-use Magento\AdobeStockAsset\Model\ResourceModel\Asset\LoadByIds;
+use Magento\AdobeStockAsset\Model\ResourceModel\Asset\Command\LoadByIds;
 use Magento\Framework\Api\AttributeValueFactory;
-use Magento\Framework\Api\Search\Document;
-use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\Api\Search\{Document,SearchResultInterface};
```
Comment (diazwatson) [inline AdobeStockAsset/Model/AppendAttributes.php]: @coderimus could you provide some documentation about this way of declare use?
Is this what you mentioned during the call?

Thanks

Comment (coderimus) [reply AdobeStockAsset/Model/AppendAttributes.php]: Hello @diazwatson ,

[In the PHP 7.0 the Group Namesapce Declaration was added ](https://wiki.php.net/rfc/group_use_declarations) This approach provides the ability to group imported namespaces a make them more structured. In the PSR-12 which approved to be used (as I know) we have some rules about this feature. You can find information here: [3. Declare Statements, Namespace, and Import Statements](https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements)

Comment (diazwatson) [reply AdobeStockAsset/Model/AppendAttributes.php]: Thanks :)

File: AdobeStockAssetApi/Model/Asset/Command/LoadByIdsInterface.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+interface LoadByIdsInterface
+{
+    /**
+     * Load Adobe Stock assets by ids
+     *
+     * @param int[] $ids
+     *
+     * @return AssetInterface[]
+     */
+    public function execute(array $ids): array;
```
Comment (sivaschenko) [inline AdobeStockAssetApi/Model/Asset/Command/LoadByIdsInterface.php]: @coderimus should we use the bulk approach to all of the Asset commands?

File: AdobeStockAssetApi/Api/CreatorRepositoryInterface.php [hunk 3]
```diff
@@ -19,46 +19,48 @@
 interface CreatorRepositoryInterface
```
Comment (sivaschenko) [inline AdobeStockAssetApi/Api/CreatorRepositoryInterface.php]: That is an API interface, please do not change the method signatures for it

File: AdobeStockAssetApi/Api/CategoryRepositoryInterface.php [hunk 4]
```diff
@@ -19,46 +19,48 @@
 interface CategoryRepositoryInterface
```
Comment (sivaschenko) [inline AdobeStockAssetApi/Api/CategoryRepositoryInterface.php]: That is an API interface, please do not change the method signatures for it

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @coderimus , please see my review comments

File: AdobeStockAssetApi/Model/Asset/Command/LoadByIdsInterface.php (hunk 2 above)
Comment (coderimus) [reply AdobeStockAssetApi/Model/Asset/Command/LoadByIdsInterface.php]: @sivaschenko I think no, we should not practise the bulk approach for all Asset related commands. From my point of view, using it everywhere is overhead and, personally, breaks the single responsibility approach. I know that this sounds like a not so good example, but in fact, it is :) When you need to get one item by its id it doesn't seem to be logical to get it with the operation for getting multiple items. We even have different conditions in the SQL (equal and in).

In conclusion: I left the LoadById command and introduced the new one - MultiplyLoadById. We need such functionality at the AppendAttributes class.

**Comment by coderimus** [comment]
@sivaschenko please, check the failed tests. It looks like all of the related to the temporary controller implementation we have for the media gallery. If yes maybe we can suppress them during the development process?

**Comment by coderimus** [comment]
@sivaschenko the majority of the failed tests fixed. There only two left, and I want to ask you for help with them. Maybe they both related to the infrastructure? Can you please check two failed tests?

**Review by sivaschenko — APPROVED** [review]
